### PR TITLE
spablob/bsc-deployment (wip)

### PR DESCRIPTION
**Pull request by Spablob**
Steps made before the PR below:

1) Deploy Opyn contracts on BSC Mainnet | `master` branch on https://github.com/opynfinance/GammaProtocol 

2) Post-deployment manual interactions
	- `Oracle` contract
		- call `setAssetPricer` with:
			- ChainlinkPricer address (0x7c117F553cF3A97e00Dc89f189Ba959C31A9260d)
			- asset address WBNB (0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c)
		- call `setStablePrice` with:
			- USDC address (0x8AC76a51cc950d9822D68b83fE1Ad97B32Cd580d)
			- uint: 100000000
	- `Whitelist` contract
		- call `whitelistCollateral` with:
			- asset address WBNB (0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c)
		- call `whitelistProduct` with:
			- underlying - asset address WBNB (0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c)
			- strike - USDC address (0x8AC76a51cc950d9822D68b83fE1Ad97B32Cd580d)
			- collateral - asset address WBNB (0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c)
			- isPut - false

### CODE REVIEW

File: scripts/deploy/30_bnb_call_theta_vault_swap.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      _managementFee: MANAGEMENT_FEE,
+      _performanceFee: PERFORMANCE_FEE,
+      _tokenName: "Ribbon BNB Theta Vault",
+      _tokenSymbol: "rBNB-THETA",
+      _optionsPremiumPricer: pricer.address,
+      _strikeSelection: strikeSelection.address,
+    },
+    {
+      isPut: false,
+      decimals: 18,
```
Comment (chuddster) [inline scripts/deploy/30_bnb_call_theta_vault_swap.ts:110]: bnb has 8 decimals

File: scripts/deploy/30_bnb_call_theta_vault_swap.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      asset: WETH_ADDRESS[chainId],
+      underlying: WETH_ADDRESS[chainId],
+      minimumSupply: BigNumber.from(10).pow(10),
+      cap: parseUnits("20000", 18),
```
Comment (chuddster) [inline scripts/deploy/30_bnb_call_theta_vault_swap.ts:114]: 8 decimals

File: test/RibbonThetaVaultWithSwap.ts [hunk 3]
```diff
@@ -115,6 +115,33 @@ describe("RibbonThetaVaultWithSwap", () => {
...
+    tokenName: "Ribbon BNB Theta Vault",
+    tokenSymbol: "rBNB-THETA",
+    asset: WETH_ADDRESS[chainId],
+    assetContractName: "IWETH",
+    strikeAsset: USDC_ADDRESS[chainId],
+    collateralAsset: WETH_ADDRESS[chainId],
+    chainlinkPricer: CHAINLINK_WETH_PRICER[chainId],
+    deltaFirstOption: BigNumber.from("1000"),
+    deltaSecondOption: BigNumber.from("1000"),
+    deltaStep: getDeltaStep("WETH"),
```
Comment (chuddster) [inline test/RibbonThetaVaultWithSwap.ts:130]: we should add a `chainId === CHAINID.BINANCE` check in [getDeltaStep](https://github.com/ribbon-finance/ribbon-v2/blob/2b7c7e2c7958ccd6b3376e80cea1a77f821f0b1c/test/helpers/utils.ts#L699)`

File: test/RibbonThetaVaultWithSwap.ts [hunk 4]
```diff
@@ -115,6 +115,33 @@ describe("RibbonThetaVaultWithSwap", () => {
...
+    depositAmount: parseEther("1"),
+    minimumSupply: BigNumber.from(10).pow(10).toString(),
+    expectedMintAmount: BigNumber.from("100000000"),
+    managementFee: BigNumber.from("2000000"),
+    performanceFee: BigNumber.from("20000000"),
+    tokenDecimals: 18,
```
Comment (chuddster) [inline test/RibbonThetaVaultWithSwap.ts:136]: 8 decimals

File: hardhat.config.ts [hunk 5]
```diff
@@ -1,143 +1,158 @@
...
+      43114: "0x422f7Bb366608723c8fe61Ac6D923023dCCBC3d7",
+      43113: "0x422f7Bb366608723c8fe61Ac6D923023dCCBC3d7",
+      1313161554: "0x46B4E6143Fb6ded2e5FBd87887Ef4f50f716dcA0",
+      1313161555: "0x46B4E6143Fb6ded2e5FBd87887Ef4f50f716dcA0",
+    },
+    owner: {
+      default: 0,
+      1: "0x77DA011d5314D80BE59e939c2f7EC2F702E1DCC4",
+      42: "0x92Dd37fbc36cB7260F0d2BD09F9672525a028fB8",
+      56: "0xF8368119Bb1073Cf01B841848725d81b542A4c19",
```
Comment (chuddster) [inline hardhat.config.ts:103]: can we change this to 0x77?

File: hardhat.config.ts [hunk 6]
```diff
@@ -1,143 +1,158 @@
...
+      43114: "0x939cbb6BaBAad2b0533C2CACa8a4aFEc3ae06492",
+      43113: "0x004FCF8052D3c7eCb7558ac0068882425a055528",
+      1313161554: "0x46B4E6143Fb6ded2e5FBd87887Ef4f50f716dcA0",
+      1313161555: "0x46B4E6143Fb6ded2e5FBd87887Ef4f50f716dcA0",
+    },
+    keeper: {
+      default: 0,
+      1: "0x691c87dc570563D1D0AD7Fd0bb099ED367858863",
+      42: "0x691c87dc570563D1D0AD7Fd0bb099ED367858863",
+      56: "0xF8368119Bb1073Cf01B841848725d81b542A4c19",
```
Comment (chuddster) [inline hardhat.config.ts:113]: can we change this to 0x69?

File: hardhat.config.ts [hunk 7]
```diff
@@ -1,143 +1,158 @@
...
+      43114: "0x422f7Bb366608723c8fe61Ac6D923023dCCBC3d7",
+      43113: "0x004FCF8052D3c7eCb7558ac0068882425a055528",
+      1313161554: "0xA4290C9EAe274c7A8FbC57A1E68AdC3E95E7C67e",
+      1313161555: "0xA4290C9EAe274c7A8FbC57A1E68AdC3E95E7C67e",
+    },
+    admin: {
+      default: 0,
+      1: "0x43a43D3404eaC5fA1ec4F4BB0879495D500e390b",
+      42: "0x422f7Bb366608723c8fe61Ac6D923023dCCBC3d7",
+      56: "0xF8368119Bb1073Cf01B841848725d81b542A4c19",
```
Comment (chuddster) [inline hardhat.config.ts:123]: can we change this to 0x43?

File: hardhat.config.ts [hunk 8]
```diff
@@ -1,143 +1,158 @@
...
+      43114: "0x31351f2BD9e94813BCf0cA04B5E6e2b7ceAFC7c6",
+      43113: "0x004FCF8052D3c7eCb7558ac0068882425a055528",
+      1313161554: "0x46B4E6143Fb6ded2e5FBd87887Ef4f50f716dcA0",
+      1313161555: "0x46B4E6143Fb6ded2e5FBd87887Ef4f50f716dcA0",
+    },
+    feeRecipient: {
+      default: 0,
+      1: "0xDAEada3d210D2f45874724BeEa03C7d4BBD41674", // Ribbon DAO
+      42: "0x92Dd37fbc36cB7260F0d2BD09F9672525a028fB8",
+      56: "0xF8368119Bb1073Cf01B841848725d81b542A4c19",
```
Comment (chuddster) [inline hardhat.config.ts:133]: can we change this to 0xDAE?

File: scripts/deploy/30_bnb_call_theta_vault_swap.ts [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    },
+    args: [optionId, manualVolOracle.address, underlyingOracle, stablesOracle],
+  });
+
+  console.log(`RibbonThetaVaultBNBCall pricer @ ${pricer.address}`);
+
+  // Can't verify pricer because it's compiled with 0.7.3
+
+  const strikeSelection = await deploy("StrikeSelectionBNBWithSwap", {
+    contract: "DeltaStrikeSelection",
```
[the remark this answers is not in the document]
Comment (Spablob) [reply scripts/deploy/30_bnb_call_theta_vault_swap.ts]: Addressed is [e2ac4dd](https://github.com/ribbon-finance/ribbon-v2/pull/300/commits/e2ac4ddec845507879426951ddb80e95fe601a95)

File: hardhat.config.ts (hunk 5 above)
Comment (Spablob) [reply hardhat.config.ts]: Addressed in https://github.com/ribbon-finance/ribbon-v2/pull/300/commits/e11a3cd85afed42c4bee272f44df8f9afe17bd31

File: hardhat.config.ts (hunk 6 above)
Comment (Spablob) [reply hardhat.config.ts]: Addressed in https://github.com/ribbon-finance/ribbon-v2/pull/300/commits/e11a3cd85afed42c4bee272f44df8f9afe17bd31

File: hardhat.config.ts (hunk 7 above)
Comment (Spablob) [reply hardhat.config.ts]: Addressed in https://github.com/ribbon-finance/ribbon-v2/pull/300/commits/e11a3cd85afed42c4bee272f44df8f9afe17bd31

File: hardhat.config.ts (hunk 8 above)
Comment (Spablob) [reply hardhat.config.ts]: Addressed in https://github.com/ribbon-finance/ribbon-v2/pull/300/commits/e11a3cd85afed42c4bee272f44df8f9afe17bd31

**Pull request closed without merging**